src/BigT/RowSort.java

```diff
@@ -0,0 +1,158 @@
+package BigT;
+
+import btree.*;
+import bufmgr.*;
+import global.*;
+import heap.*;
+import iterator.*;
+
+import java.io.IOException;
+
+public class RowSort{
+
+
+    private bigt resultTable;
+    private int numBuf;
+    private bigt sourceTable;
+    private String sourceTableName;
+    private String ColumnName;
+    private MapOrder mapOrder;
+
+    public RowSort(String sourceTableName, String resultTable, int rowOrder, String ColumnName, int n_pages)
+            throws Exception{
+        this.numBuf = n_pages;
+        this.sourceTableName = sourceTableName;
+        this.ColumnName = ColumnName;
+        this.sourceTable = new bigt(sourceTableName, 2);// insert type false
+        this.resultTable = new bigt(resultTable, 2); // insert type false
+
+        if(rowOrder == 1){
+            mapOrder = new MapOrder(MapOrder.Ascending);
+        }else if(rowOrder == 2){
+            mapOrder = new MapOrder(MapOrder.Descending);
+        }
+    }
+
+    public void run() throws Exception {
+        //create heap file with all rows recent timestamps sorted based on row + timestamp
+        Stream stream1 = new Stream(sourceTableName,  3, "*","*", "*", numBuf);
+        Heapfile rows = new Heapfile("rows");
+        createHeap(stream1, rows);
+        stream1.closestream();
+
+        Stream stream2 = new Stream(sourceTableName,  3, "*", ColumnName, "*", numBuf);
+        Heapfile filterRows = new Heapfile("filterRows");
```

Comment:
Wouldn't just running a stream on the source table with the column label to be filtered out and inserting those into a heap file of a new table suffice?